core/tests/unified_scheduler.rs

```diff
@@ -185,3 +203,97 @@ fn test_scheduler_waited_by_drop_bank_service() {
     // the scheduler used by the pruned_bank have been returned now.
     assert_eq!(pool_raw.pooled_scheduler_count(), 1);
 }
+
+#[test]
+fn test_scheduler_producing_blocks() {
```

Comment:
this test is the only code which calls some newly added fns. so, there's some `#[allow(dead_code)]` here and there across this pr.